src/fs/directory.rs

```diff
@@ -19,7 +27,13 @@ pub async fn create_dir<P: AsRef<Path>>(path: P) -> io::Result<()> {
     Op::make_dir(path.as_ref())?.await
 }
 
-/// Removes an empty directory.
+/// Removes a directory on the local filesystem.
+/// Returns a future which resolves to unit on success, and ['std::io::Error'] on failure.
+///
+/// Preconditions:
```

Comment:
I'd generally prefer indicating these as modes of failure (see the docs in tokio)